src/liquidation.rs

```diff
@@ -0,0 +1,259 @@
+        adjusted_borrow,
+        user_collateral_share
+    );
+
+    Ok(adjusted_borrow)
+}
+
+pub async fn check_liquidations(
+    sender: mpsc::Sender<Event>,
+    params: &Parameters<ClientTypeWs>,
+    min_borrow_part: U256,
+) -> anyhow::Result<()> {
+    let one_e_13 = U256::from(10000000000000_u64);
+    let one_e_18 = U256::from(1000000000000000000_u64);
+    let min_borrow_part_in_usd = min_borrow_part / one_e_18;
+
+    loggy::debug!("Checking Positions...");
+    let (total_token, total_borrow, exchange_rate, positions) = params
+        .cauldron_liquidator
+        .get_positions(
+            params.bentobox.address(),
+            params.collateral.address(),
+            params.cauldron.address(),
+            params.oracle_data.clone(),
+            params.users.clone(),
+        )
+        .call()
+        .await?
+        as (Rebase, Rebase, U256, Vec<UserPosition>);
+
+    let mut addresses_to_remove = Vec::new();
+    let mut addresses_to_liquidate = Vec::new();
+    let mut borrow_part_to_liquidate = Vec::new();
+    let mut collateral_share_to_liquidate = Vec::new();
+
+    for (i, position) in positions.iter().enumerate() {
+        loggy::debug!(
+            "borrower: {:?}, borrow_part: {}, collateral_share: {}",
+            params.users[i],
+            position.borrow_part,
+            position.collateral_share
+        );
+        if position.borrow_part.is_zero() || position.collateral_share.is_zero() {
+            addresses_to_remove.push(params.users[i]);
+            continue
+        }
+
+        let (solvent, collateral_value, borrow_value) = get_solvency_state(
+            &total_token,
+            &total_borrow,
+            exchange_rate,
+            params.collaterization_rate,
+            &position,
+        );
+
+        let collateral_value_in_usd = (collateral_value / one_e_13 / params.collaterization_rate) /
+            params.collateral_decimals;
+
+        if collateral_value_in_usd >= min_borrow_part_in_usd {
+            let ltv = (f32::from_str(&collateral_value.to_string()).unwrap() /
+                f32::from_str(&borrow_value.to_string()).unwrap()) *
+                100_f32;
+
+            if ltv - 100_f32 <= 0.1_f32 {
+                loggy::info!(
+                    "ALERT: {:?}, collateral value: {}, ltv: {:.5} %",
+                    params.users[i],
+                    Money::from_str(&collateral_value_in_usd.to_string(), iso::USD).unwrap(),
+                    ltv
+                );
+            }
+        }
+
+        if !solvent {
+            addresses_to_liquidate.push(params.users[i]);
+            borrow_part_to_liquidate.push(position.borrow_part);
+            collateral_share_to_liquidate.push(position.collateral_share);
+        }
+    }
+
+    if addresses_to_remove.len() > 0 {
+        sender.send(Event::RemoveBorrowers { addresses: addresses_to_remove }).await.unwrap();
+    }
+
+    if addresses_to_liquidate.len() > 0 {
+        liquidate(
```

Comment:
This is the part (or inside the function) that needs to be extensible to support cauldronv4 liquidation and or 0x swapper (tokens or lp etc..)

we could have a trait and have each implementation impl the trait

could be modules inside some folder named "swappers"